### DISCUSSION TITLE
Can't perform declarative mapping for SQLAlchemyBaseUserTableUUID

### DISCUSSION THREAD

**Discussion by Tur8008** [fastapi-users/fastapi-users#1159] [Q&A] [answered]
Hello,
I got a vague problem while trying to create alembic migration or just start my fastapi app.
**The exception:**

> File "/Users/tur8008/.local/share/virtualenvs/backend-J6B6n6PK/lib/python3.11/site-packages/sqlalchemy/orm/util.py", line 2306, in _extract_mapped_subtype
>     raise sa_exc.ArgumentError(
> sqlalchemy.exc.ArgumentError: Type annotation for "User.id" can't be correctly interpreted for Annotated Declarative Table form.  ORM annotations should normally make use of the ``Mapped[]`` generic type, or other ORM-compatible generic type, as a container for the actual type, which indicates the intent that the attribute is mapped. Class variables that are not intended to be mapped by the ORM should use ClassVar[].  To allow Annotated Declarative to disregard legacy annotations which don't use Mapped[] to pass, set "__allow_unmapped__ = True" on the class or a superclass this class. (Background on this error at: https://sqlalche.me/e/20/zlpr)

**The code**

```
from fastapi import Depends
from fastapi_users_db_sqlalchemy import SQLAlchemyBaseUserTableUUID, SQLAlchemyUserDatabase
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy.orm import DeclarativeMeta, declarative_base

from settings import get_async_session, engine
Base: DeclarativeMeta = declarative_base()

Base = declarative_base()

class User(SQLAlchemyBaseUserTableUUID, Base):
    pass

async def create_db_and_tables():
    async with engine.begin() as conn:
        await conn.run_sync(Base.metadata.create_all)

async def get_user_db(session: AsyncSession = Depends(get_async_session)):
    yield SQLAlchemyUserDatabase(session, User)

```

I just can't understand why it's not working -- it's just the implementation of the code from fastapi users site

<img width="1250" alt="Снимок экрана 2023-02-06 в 17 32 46" src="https://user-images.githubusercontent.com/9642686/216999220-f7c7ebd6-9cda-47f6-8ea4-311498dc77d1.png">

**Comment by frankie567** [comment, marked as answer]
You're probably running SQLAlchemy 2.0+. We do not support it yet, so you should stick on SQLAlchemy 1.4 for now.

**Reply by Tur8008** [reply]
Yes, I'm on 2.0.1 now. I'll try to downgrade it. Thanks for the answer!

**Comment by frankie567** [comment]
SQLAlchemy 2.0 support is now available in [10.4](https://github.com/fastapi-users/fastapi-users/releases/tag/v10.4.0) 😄
